frontend/src/app/offers/offer-detail/offer-detail.component.ts

```diff
@@ -4,26 +4,49 @@ import { Component, OnInit } from '@angular/core';
 import { Observable } from 'rxjs/Observable';
 
 import { environment } from '../../../environments/environment';
-import { Offer } from '../../homepage-offer/offers.model';
+import { ApiOffer } from '../../homepage-offer/offers.model';
 import { OffersService } from '../../homepage-offer/offers.service';
+import { User } from 'app/user';
+import { AuthService } from 'app/auth.service';
+import { Organization } from 'app/organization/organization.model';
+import { combineLatest } from 'rxjs/operators/combineLatest';
 
 @Component({
   selector: 'volontulo-offer-detail',
   templateUrl: './offer-detail.component.html',
   styleUrls: ['./offer-detail.component.scss'],
 })
 export class OfferDetailComponent implements OnInit {
-  public offer$: Observable<Offer>;
+  public offer$: Observable<ApiOffer>;
   public djangoRoot = environment.djangoRoot;
   public getJoinViewUrl = this.offersService.getJoinViewUrl;
+  isUserOrgMember$: Observable<boolean>;
+  user$: Observable<User | null>;
 
   constructor(
     private activatedRoute: ActivatedRoute,
+    private authService: AuthService,
     private offersService: OffersService,
   ) { }
 
   ngOnInit() {
-    this.offer$ = this.activatedRoute.params
-    .switchMap(params => this.offersService.getOffer(params.offerId));
+    this.user$ = this.authService.user$;
```

Comment:
As I promised in one of offers.service comments, the actual working solution could look something like this:
```
ngOnInit() {
    this.user$ = this.authService.user$;
    this.offer$ = this.activatedRoute.params
      .switchMap(params => this.offersService.getOffer(params.offerId))

    this.isUserOrgMember$ = this.offer$
      .pipe(combineLatest(this.user$, (offer, user): boolean => {
        if (offer && user) {
          const filteredOrganizations = user.organizations.filter(organ => organ.id === offer.organization.id);
          return filteredOrganizations.length > 0;
        }
        return false;
      }));
  }
```

Let's discuss it for a while.
FIrst of all: I didn't change the offer$ observable - it is the same as it was before. We don't need to get it from the service. It is because we are using switch map. Why? ActivatedRoute.params returns an observable, we get it's value in switchMap and call a function offersService.getOffer() which also returns the observable. SwitchMap then "removes" the observable from activatedRoute and replaces it with observable we get from getOffer(), and assigns it to this.offer$. We do not need to subscribe to this.offer$, as this is done by async pipe in component template. When the value is returned, async pipe handles it.

Later, we don't need to use if(this.offer$) as observable already exist - it has been set a few lines above. We just setup new observable isUserOrgMember$ based on this.offer$ which will resolve when this.offer$ resolves, also by using async pipe in the template. That's it. 

Check if the solution works as you intended, if not then write me on gitter, maybe I could help you some more :)